test/powershell/Language/Scripting/DeserializedTypeConversion.Tests.ps1

```diff
@@ -0,0 +1,283 @@
+﻿Describe "Tests conversion of deserialized types to original type using object properties." -Tags "CI" {
+    BeforeAll {    
+        # Create new types and test functions.
+        $type1 = Add-Type -PassThru -TypeDefinition @'
+        public class test1
+        {
+            public string name;
+            public int port;
+            public string scriptText;
+        }
+'@    
+
+        $type2 = Add-Type -PassThru -TypeDefinition @'
+        public class test2
+        {
+            private string name;
+            private int port;
+            private string scriptText;
+
+            public string Name
+            {
+                get { return name; }
+                set { name = value; }
+            }
+
+            public int Port
+            {
+                get { return port; }
+                set { port = value; }
+            }
+
+            public string ScriptText
+            {
+                get { return scriptText; }
+                set { scriptText = value; }
+            }
+        }
+'@
+
+        $type3 = Add-Type -PassThru -TypeDefinition @'
+        public class test3
+        {
+            private string name = "default";
+            private int port = 80;
+            private string scriptText = "1..6";
+
+            public string Name
+            {
+                get { return name; }
+                set { name = value; }
+            }
+
+            public int Port
+            {
+                get { return port; }
+                set { port = value; }
+            }
+
+            public string ScriptText
+            {
+                get { return scriptText; }
+            }
+        }
+'@
+
+        $type4 = Add-Type -PassThru -TypeDefinition @'
+        public class test4
+        {
+            private string name = "default";
+            private int port = 80;
+            private string scriptText = "1..6";
+
+            public string Name
+            {
+                get { return name; }
+                set { name = value; }
+            }
+
+            public int Port
+            {
+                get { return port; }
+                set { port = value; }
+            }
+
+            internal void Compute()
+            {
+                scriptText = scriptText + " Computed";
+            }
+        }
+'@
+
+        class PSClass1 {
+            [string] $name = "PSClassName1"
+            [int] $port = 80
+            [string] $scriptText = "1..6"
+        }
+
+        class PSClass2 {
+            [string] $name = "default"
+            [int] $port = 80
+            [string] $scriptText = "1..6"
+            PSClass2() {}
+            PSClass2([string] $name1, [int] $port1, [string] $scriptText1)
+            {
+                $this.name = $name1
+                $this.port = $port1
+                $this.scriptText = $scriptText1
+            }
+        }
+
+        function test-1
```

Comment:
fixed

---
In reply to: [78838599](https://github.com/Powershell/Powershell/pull/2118#discussion_r78838599) [](ancestors = 78838599)